yb_migrate/src/srcdb/source.go

```diff
@@ -31,6 +37,26 @@ type Source struct {
 	GenerateReportMode bool
 	VerboseMode        bool
 	TableList          string
+
+	sourceDB SourceDB
+}
+
+func (s *Source) DB() SourceDB {
+	if s.sourceDB != nil {
+		return s.sourceDB
+	}
+
+	s.Lock()
```

Comment:
why Lock() is needed here?

Reply:
This is a standard implementation of a Singleton pattern. If two go-routines call `source.DB()` at (almost) the same time, there is a possibility that we may instantiate the `SourceDB` twice. That's prevented by the lock.